### PR TITLE
MWPW-140041: Update merch block for release 3

### PR DESCRIPTION
**Pull request by vladen**
This PR updates `commerce.js` dependency and upgrades `merch` and `commerce` blocks for it.

Resolves: [MWPW-140041](https://jira.corp.adobe.com/browse/MWPW-140041)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/vlassenko/merch?martech=off&georouting=off
- After: https://mwpw-140041--milo--adobecom.hlx.page/drafts/vlassenko/merch?martech=off&georouting=off

### CODE REVIEW

**Comment by npeltier** [comment]
> `data-template="price"` attribute has disappeared, is this expected or it is a bug? the rest of the attributes are always shown, even if they hold a default value, e.g. `data-quantity="1"` is all over the place. I think we should also keep `data-template="price"`

i agree on both points, especially considering only price template will remain in the future

**Review by yesil — APPROVED** [review]

**Comment by yesil** [comment]
@3ch023 afaik, `data-template="price"` was never there as it is the default value.

**Review by yesil — COMMENTED** [review]
commented

**Review by honstar — APPROVED** [review]

**Comment by npeltier** [comment]
@vladen could you also make quantity = 1 the default, and remove that attribute?

File: libs/blocks/merch/merch.js [hunk 1]
```diff
@@ -79,16 +82,17 @@ export async function getPriceContext(el, params) {
   const displayRecurrence = params.get('term');
   const displayTax = params.get('tax');
   const forceTaxExclusive = params.get('exclusive');
-  const type = params.get('type');
-  const template = type === 'price' ? undefined : type;
+  let type = params.get('type');
+  if (type && type !== 'price') type = type[0].toUpperCase() + type.slice(1);
```
[the remark this answers is not in the document]
Comment (vladen) [reply libs/blocks/merch/merch.js]: @yesil good observation, thanks
I extended previous logic to also support legacy price types that could have been created with previous versions of OST.

File: test/blocks/merch/mocks/body.html [hunk 2]
```diff
@@ -39,6 +39,11 @@ <h3>Optical price</h3>
   href="/tools/ost?osi=08&type=optical&seat=true&tax=true">Price - 632B3ADD940A7FBB7864AA5AD19B8D28 - All Apps </a>
 </p>
 
+<h3>Discount pric</h3>
```
Comment (yesil) [inline test/blocks/merch/mocks/body.html:42]: typo: price

**Review by yesil — APPROVED** [review]

File: test/blocks/merch/mocks/body.html (hunk 2 above)
Comment (vladen) [reply test/blocks/merch/mocks/body.html]: thank you, fixed

**Review by npeltier — APPROVED** [review]

**Review by 3ch023 — APPROVED** [review]

**Review by Axelcureno — APPROVED** [review]

**Pull request merged**